integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,910 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+var integration = require('@segment/analytics.js-integration');
+var push = require('global-queue')('gtagDataLayer', { wrap: false });
+var Track = require('segmentio-facade').Track;
+var reject = require('reject');
+var defaults = require('@ndhoule/defaults');
+var extend = require('extend');
+
+/**
+ * Expose `GTAG`.
+ *  Purposely using different data-layer name to avoid conflicts
+ *  with any other tool.
+ */
+
+var GTAG = (module.exports = integration('Gtag')
+  .global('gtagDataLayer')
+  .option('gaWebMeasurementId', '')
+  .option('gaWebAppMeasurementId', '')
+  .option('awConversionId', '')
+  .option('dcFloodLightId', '')
+  .option('trackAllPages', false)
+  .option('trackNamedPages', true)
+  .option('trackCategorizedPages', true)
+  .option('gaOptions', {
+    classic: false,
+    enhancedEcommerce: false,
+    setAllMappedProps: true,
+    anonymizeIp: false,
+    domain: 'auto',
+    enhancedLinkAttribution: false,
+    optimize: '',
+    sampleRate: 100,
+    siteSpeedSampleRate: 1,
+    sendUserId: false,
+    useGoogleAmpClientId: false
+  })
+  .option('includeSearch', false)
+  .option('anonymizeIp', false)
+  .option('domain', 'auto')
```

Comment:
Are we using option.anonymousIp or option.domain anywhere? Or is this old and now we are using. gaOptions?

Reply:
@briemcnally `option.anonymousIp or option.domain` is GA implementation, for gatg we have segregated ga options under `gaOptions` setting. The idea is like we will create another couple of settings for floodlight and ads.